tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/builder/AbstractIncrementalBuilderTest.xtend

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2020 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.tests.builder
+
+import com.google.inject.Injector
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.util.Arrays
+import org.eclipse.n4js.N4JSGlobals
+import org.eclipse.n4js.N4JSLanguageConstants
+import org.eclipse.n4js.ide.tests.server.AbstractIdeTest
+import org.eclipse.n4js.ide.tests.server.TestWorkspaceCreator
+import org.eclipse.n4js.ide.xtext.server.ProjectStatePersisterConfig
+import org.junit.Assert
+
+/**
+ * Abstract base class for tests of the incremental builder in the LSP server.
+ */
+abstract class AbstractIncrementalBuilderTest extends AbstractIdeTest {
+
+	override protected Injector createInjector() {
+		val injector = super.createInjector();
+		val persisterConfig = injector.getInstance(ProjectStatePersisterConfig);
+		persisterConfig.setDeleteState(false);
+		persisterConfig.setWriteToDisk(true);
```

Comment:
Should this be reverted after the test?

Reply:
This default is intended for the entire test class. And we can be sure that injectors are not reused across test classes, right?

Reply:
Agreed. Usually injectors are cached by the IInjectorProvider, but given that we do super.createInjector for each method, it should be fine.